cli/command/disks/commit.go

```diff
@@ -0,0 +1,222 @@
+	return data, dcs, err
+}
+
+func assambleNewDiskRecords(dcs []*disks.DiskConfig,
+	oldDiskRecords []storage.Disk) ([]storage.Disk, []storage.Disk) {
+	keySep := ":"
+	newDiskMap := make(map[string]bool)
+
+	var newDiskRecords, diskRecordDeleteList []storage.Disk
+	for _, dc := range dcs {
+		for _, host := range dc.GetHost() {
+			key := strings.Join([]string{host, dc.GetDevice()}, keySep)
+			newDiskMap[key] = true
+			newDiskRecords = append(
+				newDiskRecords, storage.Disk{
+					Host:          host,
+					Device:        dc.GetDevice(),
+					Size:          comm.DISK_DEFAULT_NULL_SIZE,
+					URI:           comm.DISK_DEFAULT_NULL_URI,
+					MountPoint:    dc.GetMountPoint(),
+					FormatPercent: dc.GetFormatPercent(),
+					ChunkServerID: comm.DISK_DEFAULT_NULL_CHUNKSERVER_ID,
+				})
+		}
+	}
+
+	for _, dr := range oldDiskRecords {
+		key := strings.Join([]string{dr.Host, dr.Device}, keySep)
+		if _, ok := newDiskMap[key]; !ok {
+			diskRecordDeleteList = append(diskRecordDeleteList, dr)
+		}
+	}
+
+	return newDiskRecords, diskRecordDeleteList
+}
+
+func writeDiskRecord(dr storage.Disk, curveadm *cli.CurveAdm) error {
+	if diskRecords, err := curveadm.Storage().GetDisk(
+		common.DISK_QUERY_DISK_DEVICE, dr.Host, dr.Device); err != nil {
+		return err
+	} else if len(diskRecords) == 0 {
+		if err := curveadm.Storage().SetDisk(
+			dr.Host,
+			dr.Device,
+			dr.MountPoint,
+			dr.ContainerImage,
+			dr.FormatPercent); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func syncDiskRecords(data string, dcs []*disks.DiskConfig,
+	curveadm *cli.CurveAdm, options commitOptions) error {
+	oldDiskRecords := curveadm.DiskRecords()
+	tui.SortDiskRecords(oldDiskRecords)
+
+	newDiskRecords, diskRecordDeleteList := assambleNewDiskRecords(dcs, oldDiskRecords)
+	tui.SortDiskRecords(newDiskRecords)
+	oldDiskRecordsString := tui.FormatDisks(oldDiskRecords)
+	newDiskRecordsString := tui.FormatDisks(newDiskRecords)
+
+	if !options.slient {
+		diff := utils.Diff(oldDiskRecordsString, newDiskRecordsString)
+		curveadm.WriteOutln(diff)
+	}
+
+	pass := tuicomm.ConfirmYes("Disk changes are showing above. Do you want to continue?")
+	if !pass {
+		curveadm.WriteOut(tuicomm.PromptCancelOpetation("commit disk table"))
+		return errno.ERR_CANCEL_OPERATION
+	}
+
+	// write new disk records
+	for _, dr := range newDiskRecords {
+		if err := writeDiskRecord(dr, curveadm); err != nil {
+			return err
+		}
+	}
+
+	// delete obsolete disk records
+	for _, dr := range diskRecordDeleteList {
+		if dr.ChunkServerID != comm.DISK_DEFAULT_NULL_CHUNKSERVER_ID {
+			return errno.ERR_DELETE_SERVICE_BINDING_DISK.
+				F("The disk[%s:%s] is used by service[%s:%s]",
+					dr.Host, dr.Device, topology.ROLE_CHUNKSERVER, dr.ChunkServerID)
+		}
+
+		if err := curveadm.Storage().DeleteDisk(dr.Host, dr.Device); err != nil {
+			return errno.ERR_UPDATE_DISK_FAILED.E(err)
+		}
+	}
+
+	return nil
+}
+
+func runCommit(curveadm *cli.CurveAdm, options commitOptions) error {
+
```

Comment:
redundant: blank line.

Reply:
Probably we need a action for style check. I'd like to put it into practice with another pr

Reply:
Done.

Reply:
> Probably we need an action for style check. I'd like to put it into practice with another pr

That's great :)